drivers/frequency/adf4382/adf4382.h

```diff
@@ -0,0 +1,721 @@
+#define ADF4382_O_VCO_BIAS_MSK			NO_OS_BIT(2)
+#define ADF4382_O_VCO_BAND_MSK			NO_OS_BIT(1)
+#define ADF4382_O_VCO_CORE_MSK			NO_OS_BIT(0)
+
+/* ADF4382 REG004E Map */
+#define ADF4382_SPARE_4E_MSK			NO_OS_GENMASK(7, 5)
+#define ADF4382_EN_TWO_PASS_CAL_MSK		NO_OS_BIT(4)
+#define ADF4382_TWO_PASS_BAND_START_MSK		NO_OS_GENMASK(3, 0)
+
+/* ADF4382 REG004F Map */
+#define ADF4382_LUT_SCALE_MSK			NO_OS_GENMASK(7, 0)
+
+/* ADF4382 REG0050 Map */
+#define ADF4382_SPARE0_MSK			NO_OS_GENMASK(7, 0)
+
+/* ADF4382 REG0051 Map */
+#define ADF4382_SPARE1_MSK			NO_OS_GENMASK(7, 0)
+
+/* ADF4382 REG0052 Map */
+#define ADF4382_SYNC_REF_SPARE_MSK		NO_OS_GENMASK(7, 4)
+#define ADF4382_SYNC_MON_DEL_MSK		NO_OS_GENMASK(3, 0)
+
+/* ADF4382 REG0053 Map */
+#define ADF4382_SPARE_53_MSK			NO_OS_BIT(7)
+#define ADF4382_PD_SYNC_MON_MSK			NO_OS_BIT(6)
+#define ADF4382_SYNC_SEL_MSK			NO_OS_BIT(5)
+#define ADF4382_RST_SYNC_MON_MSK		NO_OS_BIT(4)
+#define ADF4382_SYNC_SH_DEL_MSK			NO_OS_GENMASK(3, 0)
+
+/* ADF4382 REG0054 Map */
+#define ADF4382_ADC_ST_CNV_MSK			NO_OS_BIT(0)
+
+#define ADF4382_SPI_3W_CFG(x)			(no_os_field_prep(ADF4382_SDO_ACTIVE_MSK, x) | \
+						 no_os_field_prep(ADF4382_SDO_ACTIVE_R_MSK, x))
+#define ADF4382_BLEED_MSB_MSK			(ADF4382_COARSE_BLEED_MSK | \
+						 ADF4382_FINE_BLEED_MSB_MSK)
+
+#define ADF4382_SPI_SCRATCHPAD_TEST		0x5A
+
+/* Specifications */
+#define ADF4382_SPI_WRITE_CMD			0x0
+#define ADF4382_SPI_READ_CMD			0x8000
+#define ADF4382_SPI_DUMMY_DATA			0x00
+#define ADF4382_BUFF_SIZE_BYTES			3
+#define ADF4382_VCO_FREQ_MIN			11000000000U	// 11GHz
+#define ADF4382_VCO_FREQ_MAX			22000000000U	// 22GHz
+#define ADF4382A_VCO_FREQ_MIN			11500000000ULL	// 11.5GHz
+#define ADF4382A_VCO_FREQ_MAX			21000000000ULL	// 21GHz
+#define ADF4382_MOD1WORD			0x2000000U	// 2^25
+#define ADF4382_MOD2WORD_MAX			0xFFFFFFU	// 2^24 - 1
+#define ADF4382_PHASE_RESYNC_MOD2WORD_MAX	0x1FFFFU	// 2^17 - 1
+#define ADF4382_CHANNEL_SPACING_MAX		78125U
+#define ADF4382_PFD_FREQ_MAX			625000000ULL	// 625MHz
+#define ADF4382_PFD_FREQ_FRAC_MAX		250000000ULL	// 250MHz
+#define ADF4382_PFD_FREQ_MIN			5400000ULL	// 5.4MHz
+#define ADF4382_DCLK_DIV1_0_MAX			160000000U	// 160MHz
+#define ADF4382_DCLK_DIV1_1_MAX			320000000U	// 320MHz
+#define ADF4382_CLKOUT_DIV_REG_VAL_MAX		4
+#define ADF4382A_CLKOUT_DIV_REG_VAL_MAX		2
+
+#define ADF4382_RFOUT_MAX			22000000000U
+#define ADF4382_RFOUT_MIN			687500000U
+#define ADF4382A_RFOUT_MAX			21000000000U
+#define ADF4382A_RFOUT_MIN			2875000000U
+#define ADF4382_REF_CLK_MAX			5000000000U
+#define ADF4382_REF_CLK_MIN			10000000
+#define ADF4382_REF_DIV_MAX			63
+#define ADF4382_OUT_PWR_MAX			15
+#define ADF4382_CPI_VAL_MAX			15
+#define ADF4382_BLEED_WORD_MAX			8191
+
+#define ADF4382_PHASE_BLEED_CNST		2044000
+#define ADF4382_VCO_CAL_CNT			202
+#define ADF4382_VCO_CAL_VTUNE			124
+#define ADF4382_VCO_CAL_ALC			250
+
+#define KHZ					1000
+#define MHZ					KHZ * KHZ
+#define S_TO_NS					1000000000U
```

Comment:
same as on linux we have `no_os_units.h`. some macro definitions can be found there. if not, please add them there.